### PR TITLE
Bug fixes in typo

### PR DESCRIPTION
**Pull request by jbuechs**
Bug fixes:
- Rendering categories page
- Create/update category
- Display error if category name is empty

Created rspec and cucumber tests for bugs

### CODE REVIEW

File: features/category.feature [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  Scenario: Successfully add new category
+    Given I am on the new category page
+    When I fill in "category_name" with "Foobar"
+    And I press "Save"
+    Then I should be on the new category page
+    And I should see "Foobar"
+    And page should have success message "Category was successfully saved."
+
+  Scenario: Error message with blank new category
```
Comment (kariabancroft) [inline features/category.feature:18]: Really nice job with these error scenarios here!

File: features/step_definitions/web_steps.rb [hunk 2]
```diff
@@ -24,6 +24,8 @@
 require File.expand_path(File.join(File.dirname(__FILE__), "..", "support", "paths"))
 require File.expand_path(File.join(File.dirname(__FILE__), "..", "support", "selectors"))
 
+require 'pry'
```
Comment (kariabancroft) [inline features/step_definitions/web_steps.rb]: eek! no commits with pry please

File: features/step_definitions/web_steps.rb [hunk 3]
```diff
@@ -276,3 +279,13 @@ def with_scope(locator)
 Then /^show me the page$/ do
   save_and_open_page
 end
+
+Then /^page should have (.+) message "([^\"]*)"$/ do |type, text|
+  page.has_css?("div.#{type}", :text => text, :visible => true)
```
Comment (kariabancroft) [inline features/step_definitions/web_steps.rb:281]: Nice use of the CSS selector to get the error messages

File: spec/controllers/admin/categories_controller_spec.rb [hunk 4]
```diff
@@ -13,7 +13,36 @@
...
+  describe "test_new" do
+    before(:each) do
+      get :new
+    end
+
+    it 'should render template new' do
+      assert_template 'new'
+    end
+
+    it 'should have empty category' do
```
Comment (kariabancroft) [inline spec/controllers/admin/categories_controller_spec.rb]: Seems like this description is not quite what you're testing for

File: spec/controllers/admin/categories_controller_spec.rb [hunk 5]
```diff
@@ -13,7 +13,36 @@
...
+      assigns(:category).should_not be_nil
+      assert !assigns(:category).valid?
+      assigns(:categories).should_not be_nil
+    end
+
+    it "successfully creates with valid params" do
+      post :new, { "category"=>{"name"=>"Foobar"} }
+      expect(flash[:notice]).to eq("Category was successfully saved.")
```
Comment (kariabancroft) [inline spec/controllers/admin/categories_controller_spec.rb:36]: Nice job testing the messages in the specs as well

File: spec/controllers/admin/categories_controller_spec.rb [hunk 6]
```diff
@@ -34,10 +63,20 @@
     end
   end
 
-  it "test_update" do
-    post :edit, :id => Factory(:category).id
-    assert_response :redirect, :action => 'index'
+  describe "test_update" do
```
Comment (kariabancroft) [inline spec/controllers/admin/categories_controller_spec.rb:66]: Thanks for adding these tests

File: features/step_definitions/web_steps.rb (hunk 2 above)
Comment (jbuechs) [inline features/step_definitions/web_steps.rb]: You would think that the multiple times this threw me into a panic during capstone would have been lesson enough. But, no.

**Comment by kariabancroft** [comment]
Great! Good to close!

**Pull request closed without merging**